students/ColeP/Session09/mailroom_oo.py

```diff
@@ -0,0 +1,185 @@
+#  in class switch dictionary stuff
+
+# def one():
+#     return 'one'
+#
+#
+# def main():
+#
+#     switch = {
+#         '1': one,
+#         '2': ...,
+#         '3': ...,
+#         '4': ...
+#     }
+#
+#     while True:
+#         choice = input('1-4')
+#         if choice not in switch.keys():
+#             print('select 1-4')
+#         else:
+#             switch[choice]()
+donors_old = {"Donny Donor": [100, 10, 45],
+          "Gav Giver": [12, 16, 20],
+          "Stingy Steve": [2, 5, 1],
+          "Freaky Frank": [1200, 999, 1005],
+          }
+
+
+class Donor:
+    def __init__(self, name, donations=None):
+        self.norm_name = self.normalize_name(name)
+        self.name = name.strip()
+
+        if donations is None:
+            self.donations = []
+        else:
+            try:
+                self.donations = list(donations)
+            except:  # wants a specific error message?
+                self.donations = [donations]
+
+        # self.ty = '''
+        # Dear {},
+        #
+        # Thank you for your most recent donation of ${:.2f} to The Org!
+        #
+        # -The Org'''.format(name, donations)
+
+    def add_donation(self, donation):
+        self.donations += [donation]
+
+    def dictionize(self):
+        return {self.name: self.donations}
+
+    def ty(self):
+        return ('''
+        Dear {},
+
+        Thank you for your most recent donation of ${:.2f} to The Org!
+
+        -The Org'''.format(self.name, self.donations[-1]))
+
+    @staticmethod
+    def normalize_name(name):
+        return name.lower().strip().replace(' ', '')
+
+
+class StatsTable:
```

Comment:
I'm a little skeptical that you need to create this class just for the stats, you should be able to do everything within the donor class itself